src/compiler/checker.ts

```diff
@@ -2370,8 +2406,11 @@ namespace ts {
             if (symbol.flags & SymbolFlags.Module) {
                 const name = (specifier.propertyName ?? specifier.name).escapedText;
                 const exportSymbol = getExportsOfSymbol(symbol).get(name);
-                markSymbolOfAliasDeclarationIfResolvesToTypeOnly(specifier, exportSymbol);
-                return resolveSymbol(exportSymbol, dontResolveAlias);
+                const resolved = resolveSymbol(exportSymbol, dontResolveAlias);
+                if (!markSymbolOfAliasDeclarationIfResolvesToTypeOnly(specifier, exportSymbol)) {

```

Comment:
this call pair happens quite a few times. I think it should be extracted, perhaps into `markSymbolOfAliasDeclarationIfResolvesToTypeOnly` itself. I can't tell if it's possible and easy to regularize the other calls to accommodate the 1-export 2-resolved pattern.

Reply:
I had the same thought when reviewing my own PR this morning, but stopped when I couldn’t come up with a satisfying name to add more logic to a function that already has a monstrous name. Any suggestions?

Reply:
Combined all three type-only checks (1. declaration is marked type-only; 2. immediate alias is marked type-only; 3. final alias is marked type-only) into one `markSymbolOfAliasDeclarationIfTypeOnly` call